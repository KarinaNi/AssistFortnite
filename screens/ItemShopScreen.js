import React from 'react';
import { View } from 'react-native';

export default class ItemShopScreen extends React.Component {
  static navigationOptions = {
    title: 'Item Shop',
  };

  render() {
    return (
      <View></View>
    )
  }
}
