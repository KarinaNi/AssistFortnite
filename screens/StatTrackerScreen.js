import React from 'react';
import { View, ImageBackground, Text, ActivityIndicator } from 'react-native';
import { getUserStats, getUserID } from '../api';
import ButtonBar from '../components/ButtonBar'
import styles from '../styles';
import background from '../assets/images/background.jpg'


export default class StatTrackerScreen extends React.Component {
  static navigationOptions = {
    title: 'Stat Tracker',
  };

  constructor(props) {
    super(props);
    this.state = {
      username: '',
      platformName: '',
      stats: {},
      loading: true,
      mode: 'solo',
      time: 'all'
    };
  }

  componentDidMount() {
    let stats = this.props.navigation.getParam('userStats', 'StatTrackerScreen: No statVal prop');
    if(stats == 'StatTrackerScreen: No statVal prop') {
      Alert.alert('Stat Traker Error',
        'Stat traker error please retry.',
        [
          { text: 'OK', onPress: () =>  this.props.navigation.goBack() },
        ],
        { cancelable: false })
      this.setState({ loading: false });
      return;
    }
    this.setState({ stats: stats, loading: false});
  }

  getTotalGames() {
    if (this.state.mode === 'solo') {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p2 == undefined) {
          return 'No solo data';
        } else {
          return this.state.stats.stats.p2.matches.displayValue;
        }
      }
      else {
        if (this.state.stats.stats.curr_p2 == undefined) {
          return 'No solo data';
        } else {
          return this.state.stats.stats.curr_p2.matches.displayValue;
        }
      }
    } else if (this.state.mode === 'duo') {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p10 == undefined) {
          return 'No duo data';
        } else {
          return this.state.stats.stats.p10.matches.displayValue;
        }
      }
      else {
        if (this.state.stats.stats.curr_p10 == undefined) {
          return 'No duo data';
        } else {
          return this.state.stats.stats.curr_p10.matches.displayValue;
        }
      }
    } else {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p9 == undefined) {
          return 'No squad data';
        } else {
          return this.state.stats.stats.p9.matches.displayValue;
        }
      }
      else {
        if (this.state.stats.stats.curr_p9 == undefined) {
          return 'No squad data';
        } else {
          return this.state.stats.stats.curr_p9.matches.displayValue;
        }
      }
    }
  }

  getTotalKills() {
    if (this.state.mode === 'solo') {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p2 == undefined) {
          return 'No solo data';
        } else {
          return this.state.stats.stats.p2.kills.displayValue;
        }
      }
      else {
        if (this.state.stats.stats.curr_p2 == undefined) {
          return 'No solo data';
        } else {
          return this.state.stats.stats.curr_p2.kills.displayValue;
        }
      }
    } else if (this.state.mode === 'duo') {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p10 == undefined) {
          return 'No duo data';
        } else {
          return this.state.stats.stats.p10.kills.displayValue;
        }
      }
      else {
        if (this.state.stats.stats.curr_p10 == undefined) {
          return 'No duo data';
        } else {
          return this.state.stats.stats.curr_p10.kills.displayValue;
        }
      }
    } else {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p9 == undefined) {
          return 'No squad data';
        } else {
          return this.state.stats.stats.p9.kills.displayValue;
        }
      }
      else {
        if (this.state.stats.stats.curr_p9 == undefined) {
          return 'No squad data';
        } else {
          return this.state.stats.stats.curr_p9.kills.displayValue;
        }
      }
    }
  }

  getWinRate() {
    if (this.state.mode === 'solo') {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p2 == undefined) {
          return 'No solo data';
        } else {
          return this.state.stats.stats.p2.winRatio.displayValue + '%';
        }
      }
      else {
        if (this.state.stats.stats.curr_p2 == undefined) {
          return 'No solo data';
        } else {
          return this.state.stats.stats.curr_p2.winRatio.displayValue + '%';
        }
      }
    } else if (this.state.mode === 'duo') {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p10 == undefined) {
          return 'No duo data';
        } else {
          return this.state.stats.stats.p10.winRatio.displayValue + '%';
        }
      }
      else {
        if (this.state.stats.stats.curr_p10 == undefined) {
          return 'No duo data';
        } else {
          return this.state.stats.stats.curr_p10.winRatio.displayValue + '%';
        }
      }
    } else {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p9 == undefined) {
          return 'No squad data';
        } else {
          return this.state.stats.stats.p9.winRatio.displayValue + '%';
        }
      }
      else {
        if (this.state.stats.stats.curr_p9 == undefined) {
          return 'No squad data';
        } else {
          return this.state.stats.stats.curr_p9.winRatio.displayValue + '%';
        }
      }
    }
  }

  getKD() {
    if (this.state.mode === 'solo') {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p2 == undefined) {
          return 'No solo data';
        } else {
          return this.state.stats.stats.p2.kd.displayValue;
        }
      }
      else {
        if (this.state.stats.stats.curr_p2 == undefined) {
          return 'No solo data';
        } else {
          return this.state.stats.stats.curr_p2.kd.displayValue;
        }
      }
    } else if (this.state.mode === 'duo') {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p10 == undefined) {
          return 'No duo data';
        } else {
          return this.state.stats.stats.p10.kd.displayValue;
        }
      }
      else {
        if (this.state.stats.stats.curr_p10 == undefined) {
          return 'No duo data';
        } else {
          return this.state.stats.stats.curr_p10.kd.displayValue;
        }
      }
    } else {
      if(this.state.time === 'all') {
        if (this.state.stats.stats.p9 == undefined) {
          return 'No squad data';
        } else {
          return this.state.stats.stats.p9.kd.displayValue;
        }
      }
      else {
        if (this.state.stats.stats.curr_p9 == undefined) {
          return 'No squad data';
        } else {
          return this.state.stats.stats.curr_p9.kd.displayValue;
        }
      }
    }
  }

  handleModeButtonPress(mode) {
    this.setState({ mode: mode });
  }

  handleTimeButtonPress(time) {
    this.setState({ time: time });
  }

  render() {
    return (
      <View style={styles.container}>
        <ImageBackground source={background} style={{ flex: 1 }}>
          {this.state && !this.state.loading ?
            <View style={{ flex: .95, flexDirection: 'column', borderRadius: 5, backgroundColor: 'rgba(43, 75, 149, 0.25)', margin: 15, padding:5, justifyContent: 'space-evenly'}}>

              <View style={{ flexDirection: 'row', justifyContent: 'center' }}>
                <Text style={[styles.subTitleTextStyle, { borderBottomWidth: 2, borderColor: '#9d4dbb' }]}>{this.state.stats.epicUserHandle} on {this.state.stats.platformNameLong}</Text>
              </View>

              <ButtonBar handleTimeButtonPress={this.handleTimeButtonPress.bind(this)} handleModeButtonPress={this.handleModeButtonPress.bind(this)} />

              <View style={{ flex: .8, flexDirection: 'column', marginLeft: 10, justifyContent: 'space-evenly' }}>
                <Text style={styles.subTitleTextStyle}>Total Games Played:</Text>
                <Text style={[styles.subTitleTextStyle, { alignSelf: 'center' }]}>{this.getTotalGames()}</Text>
                <Text style={styles.subTitleTextStyle}>Total Kills:</Text>
                <Text style={[styles.subTitleTextStyle, { alignSelf: 'center' }]}>{this.getTotalKills()}</Text>
                <Text style={styles.subTitleTextStyle}>Win Rate:</Text>
                <Text style={[styles.subTitleTextStyle, { alignSelf: 'center' }]}>{this.getWinRate()}</Text>
                <Text style={styles.subTitleTextStyle}>Kill/Death Ratio:</Text>
                <Text style={[styles.subTitleTextStyle, { alignSelf: 'center' }]}>{this.getKD()}</Text>
              </View>

            </View>
            : <View style={styles.activityIndicatorStyle}>
              <ActivityIndicator size='large' color='#152D53' />
            </View>}
        </ImageBackground>
      </View>
    );
  }
}
