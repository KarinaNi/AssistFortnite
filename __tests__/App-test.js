'use strict';

import React from 'react';
import HomeScreen from '../screens/HomeScreen'
import StatTrackerScreen from '../screens/StatTrackerScreen'
import ItemShopScreen from '../screens/ItemShopScreen'
import renderer from 'react-test-renderer';
import userStats from '../assets/__mockAPI'

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });  

//Homescreen.js __Tests__
describe('homescreen tests', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot()
  });
  it('renders the ActivityIndicator component', () => {
    const ActivityIndicator = require('ActivityIndicator');
    const tree = renderer
      .create(<ActivityIndicator animating={true} size="large" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the Background component', done => {
    const Image = require('Image');
    Image.getSize('path.jpg', (width, height) => {
      const tree = renderer.create(<Image style={{height, width}} />).toJSON();
      expect(tree).toMatchSnapshot();
      done();
    });
  });
})

//StatsTrackerScreen.js __Tests__
describe('stats screen tests', () => {
  it('renders correctly', () => {
    const navigation = { 
      getParam: (key, fallback) => {
        if(key == 'username') return 'foo'
        if(key == 'platformName') return 'foo'
        if(key == 'userStats') return userStats
      } 
    }
    const tree = renderer.create(<StatTrackerScreen navigation={navigation}/>).toJSON();
    expect(tree).toMatchSnapshot();

  });
})

//ItemShopScreen.js __Tests__
describe('item shop screen tests', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ItemShopScreen />).toJSON();
    expect(tree).toMatchSnapshot();

  });
})
