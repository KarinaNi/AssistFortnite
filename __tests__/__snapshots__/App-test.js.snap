// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homescreen tests renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1F1F1F",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(43, 75, 149, 0.25)",
          "borderRadius": 5,
          "flex": 0.3,
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Stat Tracker
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0.85,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "margin": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Enter Username Here"
              placeholderTextColor="#ffffff"
              style={
                Object {
                  "borderColor": "#9d4dbb",
                  "borderRadius": 5,
                  "borderWidth": 2,
                  "color": "#ffffff",
                  "flex": 1,
                  "margin": 3,
                  "paddingLeft": 10,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Object {
                "borderColor": "#9d4dbb",
                "borderRadius": 5,
                "borderWidth": 2,
                "flex": 1,
                "margin": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <RCTPicker
                items={
                  Array [
                    Object {
                      "label": "PC",
                      "textColor": 4278190080,
                      "value": "pc",
                    },
                    Object {
                      "label": "Xbox",
                      "textColor": 4278190080,
                      "value": "xb1",
                    },
                    Object {
                      "label": "Playstation",
                      "textColor": 4278190080,
                      "value": "ps4",
                    },
                    Object {
                      "label": "Nintendo",
                      "textColor": 4278190080,
                      "value": "nintendo",
                    },
                    Object {
                      "label": "Mobile",
                      "textColor": 4278190080,
                      "value": "touch",
                    },
                  ]
                }
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                selectedIndex={0}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#9d4dbb",
              "borderRadius": 5,
              "borderWidth": 2,
              "flex": 0.25,
              "margin": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 15,
                  "fontWeight": "bold",
                }
              }
            >
              Track
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.3,
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(43, 75, 149, 0.25)",
            "borderColor": "#9d4dbb",
            "borderRadius": 5,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 15,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Weekly Challenges
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(43, 75, 149, 0.25)",
            "borderColor": "#9d4dbb",
            "borderRadius": 5,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 15,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Random Drop
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.3,
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(43, 75, 149, 0.25)",
              "borderColor": "#9d4dbb",
              "borderRadius": 5,
              "borderWidth": 2,
              "flex": 1,
              "justifyContent": "center",
              "margin": 15,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Upcoming Items
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(43, 75, 149, 0.25)",
              "borderColor": "#9d4dbb",
              "borderRadius": 5,
              "borderWidth": 2,
              "flex": 1,
              "justifyContent": "center",
              "margin": 15,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Item Shop
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.3,
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(43, 75, 149, 0.25)",
              "borderColor": "#9d4dbb",
              "borderRadius": 5,
              "borderWidth": 2,
              "flex": 1,
              "justifyContent": "center",
              "margin": 15,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Patch Notes
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(43, 75, 149, 0.25)",
              "borderColor": "#9d4dbb",
              "borderRadius": 5,
              "borderWidth": 2,
              "flex": 1,
              "justifyContent": "center",
              "margin": 15,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Weapon Stats
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`homescreen tests renders the ActivityIndicator component 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="large"
/>
`;

exports[`homescreen tests renders the Background component 1`] = `
<Image
  style={
    Object {
      "height": 240,
      "width": 320,
    }
  }
/>
`;

exports[`itemshop screen tests renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1F1F1F",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(43, 75, 149, 0.25)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#152D53"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;

exports[`random dropper screen tests renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1F1F1F",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(43, 75, 149, 0.25)",
            "borderRadius": 5,
            "flex": 0.3,
            "flexDirection": "column",
            "justifyContent": "center",
            "margin": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 25,
              "fontWeight": "bold",
            }
          }
        >
          Strategy
        </Text>
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 40,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(43, 75, 149, 0.25)",
            "borderRadius": 5,
            "flex": 0.3,
            "flexDirection": "column",
            "justifyContent": "center",
            "margin": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 25,
              "fontWeight": "bold",
            }
          }
        >
          Drop Location
        </Text>
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 40,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(43, 75, 149, 0.25)",
            "borderColor": "#9d4dbb",
            "borderRadius": 5,
            "borderWidth": 2,
            "flex": 0.1,
            "justifyContent": "center",
            "margin": 15,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Drop
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`stats screen tests renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1F1F1F",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(43, 75, 149, 0.25)",
          "borderRadius": 5,
          "flex": 0.95,
          "justifyContent": "space-evenly",
          "margin": 15,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 30,
                "fontWeight": "bold",
              },
              Object {
                "borderBottomWidth": 2,
                "borderColor": "#9d4dbb",
              },
            ]
          }
        >
          foo
           on 
          foo
        </Text>
      </View>
      <View
        style={
          Object {
            "borderColor": "#9d4dbb",
            "borderRadius": 10,
            "borderWidth": 3,
            "flex": 0.15,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "margin": 10,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(43, 75, 149, 0.25)",
              "borderBottomLeftRadius": 5,
              "borderTopLeftRadius": 5,
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 15,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Solo
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderColor": "#9d4dbb",
              "borderLeftWidth": 3,
              "borderRightWidth": 3,
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 15,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Duo
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 15,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Squad
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 0.75,
            "flexDirection": "column",
            "justifyContent": "space-evenly",
            "marginLeft": 10,
            "marginTop": 25,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 30,
              "fontWeight": "bold",
            }
          }
        >
          Total Games Played:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 30,
                "fontWeight": "bold",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          507
        </Text>
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 30,
              "fontWeight": "bold",
            }
          }
        >
          Win Rate:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 30,
                "fontWeight": "bold",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          0.394 %
        </Text>
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 30,
              "fontWeight": "bold",
            }
          }
        >
          Kill/Death Ratio:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 30,
                "fontWeight": "bold",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          1.081
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`weekly screen tests all weeks render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1F1F1F",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`weekly screen tests each week screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1F1F1F",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
